### ISSUE TITLE
Payload should be pure JSON not a String map

### ISSUE THREAD

**Issue by mneumann** [GildedHonour/frank_jwt#3]
At the moment, it is not possible to create a payload of, for instance:

```
{
"sub": "1234567890",
"name": "John Doe",
"admin": true
"exp": 123545679
}
```

**Comment by GildedHonour** [comment]
As far as I'm concerned, in other libraries it's the same. Why not to create a payload out of the keys and values of that json?

**Comment by gsingh93** [comment]
Why not use the standard `Json` type from [rustc_serialize](https://github.com/rust-lang/rustc-serialize) instead of a `BTreeMap`?

**Comment by GildedHonour** [comment]
@gsingh93 what are the advantages?

**Comment by gsingh93** [comment]
A strongly typed API instead of a "stringly" typed API. For example, bools are actual bools, not just strings that can be "true" or "false".

[rust-jwt](https://github.com/mikkyang/rust-jwt) does this.

**Comment by GildedHonour** [comment]
Ok

**Comment by rrichardson** [comment]
This is addressed by #16
